### PR TITLE
feat(options): add icon hiding/appearing on attribute change

### PR DESCRIPTION
**Pull request by nattallius**
#351 
Any variables naming suggestions are welcomed :)

### CODE REVIEW

File: src/plugins/header/options/options.ts [hunk 1]
```diff
@@ -43,10 +49,24 @@ export class UIPOptions extends UIPPlugin {
...
+      const option = this.options.get(config.rootControlledAttr);
+      if (!option) return;
+      this.options.delete(config.rootControlledAttr);
+      option.remove();
+    }
+  }
+
+  protected render(options = UIPOptions.UIPOptionsConfig) {
+    options = options.filter(option => !option.canActivate || option.canActivate(this));
+    options.forEach((option) => this.options.set(option.rootControlledAttr, UIPOption.create(option)));
```
Comment (yadamskaya) [inline src/plugins/header/options/options.ts:69]: Minor:
```suggestion
  protected render(options: OptionConfig[] = UIPOptions.UIPOptionsConfig): void {
    options = options.filter(option => !option.canActivate || option.canActivate(this));
    options.forEach(option => this.options.set(option.rootControlledAttr, UIPOption.create(option)));
```

File: src/plugins/header/options/options.ts [hunk 2]
```diff
@@ -43,10 +49,24 @@ export class UIPOptions extends UIPPlugin {
...
   }
 
-  protected render() {
-    const options = UIPOptions.UIPOptionsConfig.filter(option => !option.canActivate || option.canActivate(this));
-    const entries: [string, UIPOption][] = options.map(option => [option.attribute, UIPOption.create(option)]);
-    this.options = new Map(entries);
+  protected attributeChangedCallback(attrName: string, oldVal: string | null, newVal: string | null) {
+    if (oldVal === newVal || !this.options) return;
+
+    const config = UIPOptions.UIPOptionsConfig.find((elem) => elem.attrName === attrName);
```
Comment (yadamskaya) [inline src/plugins/header/options/options.ts:55]: Is there a way to simplify this method? But mb it looks a bit complicated just to me..

Comment (yadamskaya) [inline src/plugins/header/options/options.ts:55]: And here I would do like this:
```suggestion
    const config: OptionConfig | undefined = UIPOptions.UIPOptionsConfig.find(elem => elem.attrName === attrName);
```

**Review by yadamskaya — APPROVED** [review]

**Review by Sisha3342 — APPROVED** [review]

File: src/plugins/header/options/options.ts [hunk 3]
```diff
@@ -54,19 +54,19 @@ export class UIPOptions extends UIPPlugin {
...
       this.render([config]);
-    } else {
-      const option = this.options.get(config.rootControlledAttr);
+    } else { // remove option icon
+      const option = this.options.get(config.optionValue);
       if (!option) return;
-      this.options.delete(config.rootControlledAttr);
+      this.options.delete(config.optionValue);
       option.remove();
     }
```
Comment (yadamskaya) [inline src/plugins/header/options/options.ts:64]: Let's do it like this 
```
const config: OptionConfig | undefined = UIPOptions.UIPOptionsConfig.find(elem => elem.attrName === attrName);
    if (!config) return;

    if (newVal === null) this.render([config]); // add option icon
    if (newVal !== null) { // remove option icon
      const option = this.options.get(config.optionValue);
      if (!option) return;
      this.options.delete(config.optionValue);
      option.remove();
    }
```
And btw the name `optionValue` for OptionConfig is very nice to me!

File: src/plugins/header/options/option/option.ts [hunk 4]
```diff
@@ -18,18 +19,33 @@ export type OptionConfig = {
 export class UIPOption extends ESLBaseElement {
   static is = 'uip-option';
   @attr() public attribute: string;
-  @boolAttr() public active: boolean;
+  public _active: boolean;
```
[the remark this answers is not in the document]
Comment (nattallius) [reply src/plugins/header/options/option/option.ts:22]: There was till Alex helped me with better approach)

**Review by Sisha3342 — APPROVED** [review]

**Review by AlexanderBazukevich — APPROVED** [review]

**Pull request merged**